src/screens/SignUp/index.tsx

```diff
@@ -1,12 +1,138 @@
-import React from 'react';
-import {Text, View} from 'react-native';
+import React, {useState, useCallback} from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  KeyboardAvoidingView,
+  Platform,
+  ScrollView,
+  Pressable,
+  Image,
+} from 'react-native';
+
+import {SafeAreaView} from 'react-native-safe-area-context';
+import {TextInput} from 'react-native-gesture-handler';
+import {commonStyle} from '~/src/styles/commonStyle';
+import {SignUpScreenProps} from '~/src/types/type';
+import {theme} from '~/src/styles/theme';
+
+const SignUp = ({navigation}: SignUpScreenProps) => {
+  const [hiddenPw, setHiddenPw] = useState(true);
+  const [hiddenPwCheck, setHiddenPwCheck] = useState(true);
+
+  const hiddenPwHandler = useCallback(isHidden => {
```

Comment:
1. 여기는 `useCallback()`이 굳이 필요 없는 것 같습니다.
공식문서에 보시면
```
This is useful when passing callbacks to optimized child components that rely on reference equality to prevent unnecessary renders 
```
라고 나와있는데, 지금 컴포넌트에는 child component도 없고, state가 바뀔 때 이미지 source만 바뀌는 거기 때문에 unnecessary render도 없을 것 같습니다. 
- `useCallback()`을 사용하고자 디펜던시를 추가하신다면 가져오는 이미지가 들어가야합니다.  이미지가 로드되기 전에 해당 함수를 실행하면 에러가 발생할테니까요.

3. ES6 문법에 맞게 `require`대신 `import`문을 사용해주세요.

Reply:
1,2. 알려주신 내용을 바탕으로 구글링하여 useCallback에 대해 더 이해할 수 있었습니다. 감사합니다!! 
- 함수 생성부분에 대해 한 가지 더 궁금한 게 있습니다. 만약 여러 State가 있고 그 값이 바뀔때마다 함수컴포넌트 전체가 재렌더링이 된다면 hiddenPwHandler 함수도 재렌더링 됨과 동시에 재생성되는 것으로 알고 있는데, 이경우엔 선언만 되고 메모리를 많이 차지하지 않기 때문에 말씀처럼 필요한 경우가 아니라면 굳이 useCallback을 남용할 필요가 없는걸까요?

3. require 대신 import 문 사용
- 모듈을 불러올 수 있도록 index.d.ts 파일 생성, `declare module '*.png';` 을 작성하고, 아래와 같이 회원가입 index.tsx 로직을 수정하였습니다.
 
```tsx
import eyeOff from 'assets/images/Icon_feather_eye_off.png';
import eyeOn from 'assets/images/Icon_feather_eye_on.png';

  const hiddenPwHandler = (isHidden: boolean) => {
    return isHidden ? eyeOff : eyeOn;
  };
```